README.md

```diff
@@ -0,0 +1,125 @@
+# TAMS Master Project 2022/2023
+
+This repository contains the codebase for a [Marimba](https://en.wikipedia.org/wiki/Marimba) playing robot developed during the TAMS Master Project 2022/2023. 
+It is currently still "work in progress".
+
+## Setup 
+
+The robot uses Ubuntu 20.04 and ROS noetic. 
+So make sure that it is installed on your system.
+Visit the [ROS Wiki](http://wiki.ros.org/noetic/Installation) for further details.
+
+Remember to source the ROS noetic installation in EACH terminal you use or add it once to your `.bashrc` file like this:
+
+```bash
+echo "source /opt/ros/noetic/setup.bash" >> ~/.bashrc
+source ~/.bashrc
+```
+
+Make sure to [setup your ssh key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account) for easy synchronization without authorization.
+
+
+Run the following commands:
+
+```bash
+# Create catkin workspace folder
+mkdir -p catkin_ws/src
+
+# Clone repository into source folder
+cd catkin_ws/src
+git clone git@github.com:UHHRobotics22-23/robot_project 
+
+# Initilize workspace
+catkin init
+
+# Setup rosdep to download dependencies (only needed once after installing ROS)
+sudo rosdep init
+
+# Update rosdep packages list
+rosdep update
+
+# Install all dependencies needed by our repository
+rosdep install --from-paths src --ignore-src -r -y
+
+# Install python dependencies
+pip3 install -r requirements.txt
+```
+
+Now you should be ready to go.
+
+## Update workspace
+
+You might want to update the state of the software and get recent changes.
+Normally a simple pull should be enough.
+
+```bash
+# Go to the repository
+cd catkin_ws/src/robot_project
+
+# Pull recent changes
+git pull
```

Comment:
I don't think that should be part of this.
*Iff*: it would make sense for me to mention a branch that should be updated. I guess that is main here.

Reply:
but we can leave it as it is

Reply:
This just updates the current branch you are on. Yes, the `git pull` itself might be a bit trivial, but I mainly have this part for dependency update process afterwards. So I also included the pull to leave nothing out.